src/containerapp/azext_containerapp/_utils.py

```diff
@@ -104,11 +109,6 @@ def get_linker_client(cmd):
     return linker_client
 
 
-def validate_binding_name(binding_name):
```

Comment:
[`az containerapp create --bind "my-app"` throws error when service has dashes in the name "-" · Issue #6524 · Azure/azure-cli-extensions (github.com)](https://github.com/Azure/azure-cli-extensions/issues/6524)
[{AzureContainerApp} az containerapp create --bind "my-app" throws error when service has dashes in the name "-" by navba-MSFT · Pull Request #6530 · Azure/azure-cli-extensions (github.com)](https://github.com/Azure/azure-cli-extensions/pull/6530)

Please take a look about this, bind resource with service connector with `--bind` is not accept name contains `-`

Reply:
Thanks a lot. I have added the validation logic back and added some special logic for java component